vsomeip-sys/build.rs

```diff
@@ -0,0 +1,264 @@
+        .file("src/glue/src/application_wrapper.cpp")
+        .include(&include_dir)
+        .include(&vsomeip_interface_path)
+        .include(&runtime_wrapper_dir)
+        .flag_if_supported("-Wno-deprecated-declarations") // suppress warnings from C++
+        .flag_if_supported("-Wno-unused-function") // compiler compiling vsomeip
+        .flag_if_supported("-std=c++17")
+        .extra_warnings(true)
+        .compile("cxx-portion");
+    println!("cargo:rerun-if-changed=src/cxx_bridge.rs");
+
+    // we rewrite the autocxx generated code to suppress the cargo warning about unused imports
+    let file_path = Path::new(&out_dir)
+        .join("autocxx-build-dir")
+        .join("rs")
+        .join("autocxx-ffi-default-gen.rs");
+    println!("cargo:warning=# file_path : {}", file_path.display());
+
+    if !file_path.exists() {
+        panic!("Unable to find autocxx generated code to rewrite");
+    }
+
+    // Run rustfmt on the file
+    let status = Command::new("rustfmt")
+        .arg(&file_path)
+        .status()
+        .expect("Failed to execute rustfmt");
+
+    if !status.success() {
+        panic!("Failed to format autocxx generated file");
+    }
+
+    // Open the input file for reading
+    let input_file = File::open(&file_path).expect("Failed to open the input file for reading");
+    let reader = BufReader::new(input_file);
+
+    // Create a temporary file for writing the modified content
+    let temp_file_path = file_path.with_extension("tmp");
+    let temp_file =
+        File::create(&temp_file_path).expect("Failed to create a temporary file for writing");
+    let mut writer = BufWriter::new(temp_file);
+
+    for line in reader.lines() {
+        let mut line = line.expect("Failed to read a line from the input file");
+        line = line.replace(
+            "pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete",
+            "#[allow(unused_imports)] pub use bindgen::root::std_chrono_duration_long_AutocxxConcrete"
+        );
+        line = line.replace(
+            "pub use bindgen::root::std_chrono_duration_int64_t_AutocxxConcrete;",
+            "#[allow(unused_imports)] pub use bindgen::root::std_chrono_duration_int64_t_AutocxxConcrete;"
+        );
+        line = line.replace(
+            "pub use super::super::bindgen::root::std::chrono::seconds;",
+            "",
+        );
+        line = line.replace("pub unsafe fn create_payload1", "unsafe fn create_payload1");
+        line = line.replace("successfully [de]registered", "successfully de/registered");
+        line = line.replace(
+            "#[derive(Clone, Hash, PartialEq, Eq)]",
+            "#[derive(Clone, Hash, PartialEq, Eq, Debug)]",
+        );
+        line = line.replace(
+            "pub unsafe fn set_data(self: Pin<&mut payload>, _data: *const u8, _length: u32);",
+            "pub(crate) unsafe fn set_data(self: Pin<&mut payload>, _data: *const u8, _length: u32);"
```

Comment:
why do you need to reduce visibility here?

Reply:
Broke these apart to clarify